host/host.cpp

```diff
@@ -1,33 +1,77 @@
 // Copyright (c) Open Enclave SDK contributors.
 // Licensed under the MIT License.
 
+#include <assert.h>
 #include <iostream>
-#include "../constants/network.h"
-#include "../gen-cpp/RPC.h"
+#include <openenclave/host.h>
 #include <thrift/protocol/TBinaryProtocol.h>
 #include <thrift/server/TSimpleServer.h>
 #include <thrift/transport/TServerSocket.h>
 #include <thrift/transport/TBufferTransports.h>
 
+#include "ortoa_u.h"
+#include "redis.h"
+#include "../constants/constants.h"
+#include "../constants/shared.h"
+#include "../gen-cpp/RPC.h"
+
 using namespace ::apache::thrift;
 using namespace ::apache::thrift::protocol;
 using namespace ::apache::thrift::transport;
 using namespace ::apache::thrift::server;
 
+bool check_simulate(int argc, char *argv[]) {
+  for (int i = 2; i < argc; ++i) {
+    if (strcmp(argv[i], "--simulate") == 0) return true;
+  }
+  return false;
+}
+
 class RPCHandler : virtual public RPCIf {
+ private:
+  inline static uint32_t simulate_flag = OE_ENCLAVE_FLAG_DEBUG;
+  inline static char* oe_enclave_path;
+  inline static oe_enclave_t* enclave;
+  redisCli rd;
+
  public:
   RPCHandler() {
-    // Your initialization goes here
+    oe_result_t result = oe_create_ortoa_enclave(oe_enclave_path, OE_ENCLAVE_TYPE_SGX, simulate_flag, NULL, 0, &enclave);
```

Comment:
I may be stupid but where does `simulate_flag` come from?

Reply:
nvm.

Reply:
`simulate_flag` is a static member variable of the RPCHandler class that's set when we init the host. This bypasses the discussion we had before of having to deal with passing flags from the client